### PR TITLE
feat: weather-aware tyre wear extrapolation with sliding window

### PR DESCRIPTION
**Pull request by PettableCat**
## 📝 Description

Adds weather-aware tyre wear extrapolation. The extrapolator now segments wear data by weather condition (dry vs wet) so predictions reflect current track conditions instead of averaging across weather transitions.

**Key changes:**
- `TyreWearPerLap`: new `weather_id` field + JSON serialization
- `TyreWearExtrapolator`: weather-aware segmentation (`_weather_group()`, `_filter_to_current_weather()`)
- Sliding window support (`window_size` parameter) for adapting to changing conditions
- Sequential lap indexing prevents SC gaps from diluting the regression slope
- Fallback: if the current weather segment has < 3 laps, all data is used

**Weather grouping:** Dry = {Clear, LightCloud, Overcast}, Wet = {LightRain, HeavyRain, Storm, Thunderstorm}

Fully backward compatible — legacy data without `weather_id` behaves identically to before.

---

## 📂 Type of change

- [ ] Bug fix 🐞
- [x] New feature 🚀
- [ ] Refactor 🔨
- [ ] Documentation 📚
- [x] Tests ✅
- [ ] Other:

---

### ✅ Integration Tests

* [x] All integration tests pass

```
All 48 tyre wear extrapolator tests pass (TestSimpleLinearRegression,
TestTyreWearExtrapolator, TestTyreWearExtrapolatorWithNonRacingLaps,
TestTyreWearExtrapolatorWithMissingLaps, TestTyreWearExtrapolatorWithSCGaps,
TestTyreWearExtrapolatorSlidingWindow, TestTyreWearExtrapolatorWeatherSegmentation)
```

---

## 🧱 `lib/` Directory Changes

* [x] This PR modifies or adds files under `lib/`
* [x] I have updated or added unit tests to reflect those changes

```
New test classes:
- TestTyreWearExtrapolatorWithSCGaps: SC gap regression accuracy
- TestTyreWearExtrapolatorSlidingWindow: dry/wet transitions, window edge cases
- TestTyreWearExtrapolatorWeatherSegmentation: weather grouping, fallback logic, backward compat
```

---

## 📋 General Checklist

* [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) guide
* [x] My code follows project style conventions
* [x] All new and existing tests pass
* [x] I have added relevant documentation/comments
* [x] I have reviewed my changes and believe they are ready to merge

## Summary by Sourcery

Add weather-aware tyre wear extrapolation with sliding-window regression and sequential lap indexing to improve prediction accuracy under changing track conditions while preserving legacy behaviour.

New Features:
- Introduce weather-aware segmentation of tyre wear data so extrapolation uses laps from the current dry or wet weather group.
- Add an optional sliding window parameter to base regressions on the N most recent racing laps, allowing adaptation to condition changes.
- Extend TyreWearPerLap with an optional weather identifier that is included in its JSON representation.

Enhancements:
- Use sequential racing-lap indices in regressions and prediction to prevent gaps such as safety-car periods from diluting tyre wear slopes.
- Segment tyre wear data simultaneously by racing status and weather group so continuous conditions are modeled separately.

Tests:
- Add comprehensive unit tests covering SC-gap handling, sliding-window behaviour, weather-based segmentation, fallback logic, and backward compatibility for legacy data.

### CODE REVIEW

File: lib/tyre_wear_extrapolator/tyre_wear_per_lap.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ class TyreWearPerLap:
     lap_number: Optional[int] = None
     is_racing_lap: bool = True
     desc: Optional[str] = None
+    weather_id: Optional[int] = None
```
Comment (ashwin-nat) [inline lib/tyre_wear_extrapolator/tyre_wear_per_lap.py]: where is this being set from the core logic?
